File: core/src/main/java/net/rickiekarp/core/components/button/SidebarButton.java
package net.rickiekarp.core.components.button;

import javafx.scene.control.Button;

public final class SidebarButton extends Button {
    private final String identifier;

    public final String getIdentifier() {
        return this.identifier;
    }

    public SidebarButton(String identifier) {
        this.identifier = identifier;
        setText(identifier);
        getStyleClass().addAll("sidebar-button");
    }
}

File: AppCore/Sources/com/rkarp/appcore/components/button/SidebarButton.kt
package com.rkarp.appcore.components.button

import com.rkarp.appcore.controller.LanguageController
import javafx.scene.control.Button

/**
 * Creates a basic button using some css properties defined in a Theme.css file com.rkarp.flc.themes
 */
class SidebarButton(val identifier: String) : Button(LanguageController.getString(identifier)) {

    init {
        styleClass.addAll("sidebar-button")
    }
}